x/oracle/keeper/keeper.go

```diff
@@ -35,3 +38,40 @@ func NewKeeper(
 		paramSpace: paramSpace,
 	}
 }
+
+func (k Keeper) VerifySignature(ctx sdk.Context, msg codec.ProtoMarshaler, sigBz []byte) error {
+	bz, err := k.cdc.Marshal(msg)
+	if err != nil {
+		return err
+	}
+
+	oraclePubKeyBz := k.GetParams(ctx).MustDecodeOraclePubKey()
+	oraclePubKey, err := btcec.ParsePubKey(oraclePubKeyBz, btcec.S256())
+	if err != nil {
+		return err
+	}
+
+	signature, err := btcec.ParseSignature(sigBz, btcec.S256())
+	if err != nil {
+		return err
+	}
+
+	if !signature.Verify(bz, oraclePubKey) {
+		return fmt.Errorf("failed to signature validation")
+	}
+
+	return nil
+}
+
+func (k Keeper) VerifyOracle(ctx sdk.Context, oracleAddress string) error {
+	_, err := sdk.AccAddressFromBech32(oracleAddress)
+	if err != nil {
+		return err
+	}
+
+	// TODO Check is oracle registered?
+
+	// TODO Check is registered oracle's uniqueId correct?
+
+	return nil
+}
```

Comment:
Both TODOs make sense to me.
If there are some critical bugs in old version oracle, then only certificates issued by the upgraded version of oracle should be valid. That is, unique ID should also be added to certificate. wdyt?

If then, this function probably should check if the oracle is registered or not. The other one is validation of certificate.

Reply:
Umm, I think this function is only checks Oracle.

As you said, It seems that we need to verify the uniqueID of the certificate.
I will implement a function to verify the uniqueID of a certificate.